src/Avalonia.Visuals/Utilities/Span.cs

```diff
@@ -0,0 +1,596 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//+-----------------------------------------------------------------------
+//
+//
+//
+//  Contents:  Generic span types
+//
+//             [As of this creation, C# has no real generic type system]
```

Comment:
At this point I'm not sure if using this code is just archeology or already necromancy :D
WPF seems to be using this but I feel like we will need to do a modernization pass sometime.

Reply:
I was mainly concerned touching this because of future porting efforts

Reply:
I reckon that WPF text formatting is probably not getting too many changes anyway but yeah we can do this after we are done porting other stuff.